docs/data-product-studio/data-quality/failed-events/exploring-failed-events/file-storage/index.md

```diff
@@ -65,3 +74,274 @@ Step 4 - drill down into the folder structure by year, month, day and time
 ![](images/failed-evs-gcs-7.jpg)
 
 Step 5 - once you find the raw files you can download them and view them in a text editor
+
+  </TabItem>
+</Tabs>
+
+## Using Athena or BigQuery
+
+[Athena](https://aws.amazon.com/athena/) on AWS and [BigQuery](https://cloud.google.com/bigquery) on GCP are tools that let you query your failed events, using the cloud storage files as a back-end data source.
+
+```sql
+SELECT data.failure.messages FROM adapter_failures
+WHERE from_iso8601_timestamp(data.failure.timestamp) > timestamp '2020-04-01'
+```
+
+This approach is great for debugging your pipeline without the need to load your failed events into a separate database.
+
+Before you can query this data, you need to create corresponding tables in Athena or BigQuery as we explain below. Each different failed event type (e.g. schema violations, adapter failures) has a different schema, so you will need one table per event type.
+
+## Creating the tables
+
+<Tabs groupId="warehouse" queryString>
+  <TabItem value="athena" label="Athena" default>
```

Comment:
this is very picky but it would be great if these tabs matched up with the ones above, so if they selected GCP it shows the BQ tab here. Think it needs the same tabs groupId and values?

Reply:
Good point!

Reply:
there's still one set of tabs to update below as well, under Creating the tables

Reply:
🤦  fixed